controller/api/locking.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from models import get_etcd_client
+
+log = logging.getLogger(__name__)
+LOCK_PATH_TEMPLATE = '/deis/controller/locks/{name}'
+
+
+class AlreadyLocked(Exception):
+    pass
+
+
+def lock(name, ttl=3600):
+    key = LOCK_PATH_TEMPLATE.format(name=name)
+    value = 'locked'
+    client = get_etcd_client()
```

Comment:
then we're back to etcd or postgres (I'm positing that we don't want to add extra infrastructure here). I'm happy to prototype something if we decide to go do this. maybe something for post-beta